databricks/koalas/internal.py

```diff
@@ -398,11 +403,12 @@ def __init__(self, sdf: spark.DataFrame,
         assert isinstance(sdf, spark.DataFrame)
         if index_map is None:
             # Here is when Koalas DataFrame is created directly from Spark DataFrame.
-            assert "__index_level_0__" not in sdf.schema.names, \
-                "Default index column should not appear in columns of the Spark DataFrame"
+            assert any(SPARK_INDEX_NAME_PATTERN.match(name) for name in sdf.schema.names), \
```

Comment:
`not any(SPARK_INDEX_NAME_PATTERN.match(name) for name in sdf.schema.names)` or `all(not SPARK_INDEX_NAME_PATTERN.match(name) for name in sdf.schema.names)`?